components/setup/BuildInstruction.tsx

```diff
@@ -34,22 +36,51 @@ const BuildInstruction: React.FC<BuildInstructionProps> = ({ selectedHosting, us
 	const [isRepoSelectionDone, setIsRepoSelectionDone] = useState<boolean>(false);
 	const [installId, setInstallId] = useState<string | null>(null);
 	const [isGetInstallIdLoading, setIsGetInstallIdLoading] = useState(false);
+	const [handleGithubPatInputValue, setHandleGithubPatInputValue] = useState<string>("");
+	const [errorMessage, setErrorMessage] = useState<string | null>(null);
+	const [maskedGithubPat, setMaskedGithubPat] = useState('');
+	const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+	const handleGithubPatInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setHandleGithubPatInputValue(event.target.value);
+    };
+
+	const encrypt_github_pat = (handleGithubPatInputValue: string) => {
+		let github_pat_encryption_secret_key = process.env.NEXT_PUBLIC_GITHUB_PAT_ENCRYPTION_SECRET_KEY;
+		if (github_pat_encryption_secret_key) {
+			return encrypt(github_pat_encryption_secret_key, handleGithubPatInputValue);
+		}
+	}
+
+	const maskGithubPat = (handleGithubPatInputValue: string) => {
+		return handleGithubPatInputValue.replace(/.(?=.{4})/g, '*');  // Mask all but the last four characters
+	};
 
 	const handleBuildButtonClick = () => {
 		setIsTriggerBuildButtonDisabled(true);
 		setBuildStatus(null);
+		setErrorMessage(null);
+		let encrypted_github_pat = encrypt_github_pat(handleGithubPatInputValue);
 
-		axios.post('/api/dpu/trigger', { userId })
+		axios.post('/api/dpu/trigger', { userId, selectedHosting, selectedInstallationType, selectedProvider, github_pat: encrypted_github_pat })
 			.then((response) => {
 				console.log('[handleBuildButtonClick] /api/dpu/trigger response:', response.data);
 				setBuildStatus(response.data);
+				setIsTriggerBuildButtonDisabled(false);
+				if (!response.data.success) {
+					setErrorMessage('Failed to trigger build: ' + response.data.message); // Handle backend-specific error messages
+				}
 				if (response.data.success) {
+					setMaskedGithubPat(maskGithubPat(handleGithubPatInputValue)); // Mask the GitHub PAT
+					setIsInputDisabled(true);
```

Comment:
Shouldn't this be disabled as soon as the trigger button is clicked? (Around line no 62)

Reply:
yeah, I had thought of it but wasn't sure.